dashboard/client/src/pages/node/NodeRow.tsx

```diff
@@ -0,0 +1,272 @@
+            {memoryConverter(disk["/"].used)}/{memoryConverter(disk["/"].total)}
+            ({disk["/"].percent.toFixed(1)}%)
+          </PercentageBar>
+        )}
+      </TableCell>
+      <TableCell align="center">{memoryConverter(networkSpeed[0])}/s</TableCell>
+      <TableCell align="center">{memoryConverter(networkSpeed[1])}/s</TableCell>
+      <TableCell>
+        <Link to={`/log/${encodeURIComponent(logUrl)}`}>Log</Link>
+      </TableCell>
+    </TableRow>
+  );
+};
+
+type WorkerRowProps = {
+  /**
+   * Details of the worker
+   */
+  worker: Worker;
+  /**
+   * Detail of the node the worker is inside.
+   */
+  node: NodeDetail;
+};
+
+/**
+ * A single row that represents the data of a Worker
+ */
+const WorkerRow = ({ node, worker }: WorkerRowProps) => {
+  const classes = useNodeRowStyles();
+
+  const { mem, logUrl } = node;
+  const {
+    pid,
+    cpuPercent: cpu = 0,
+    memoryInfo,
+    coreWorkerStats,
+    cmdline,
+  } = worker;
+
+  const coreWorker = coreWorkerStats.length ? coreWorkerStats[0] : undefined;
+  const workerLogUrl = coreWorker
+    ? `/log/${encodeURIComponent(logUrl)}?fileName=${coreWorker.workerId}`
+    : `/log/${encodeURIComponent(logUrl)}`;
+
+  return (
+    <TableRow key={pid}>
+      <TableCell></TableCell>
+      <TableCell>
+        <StatusChip type="worker" status="ALIVE" />
+      </TableCell>
+      <TableCell align="center">
+        {coreWorker && (
+          <Tooltip title={coreWorker.workerId} arrow interactive>
+            <span className={classes.idCol}>{coreWorker.workerId}</span>
+          </Tooltip>
+        )}
+      </TableCell>
+      <TableCell align="center">{cmdline[0]}</TableCell>
+      <TableCell align="center">{pid}</TableCell>
+      <TableCell>
+        <PercentageBar num={Number(cpu)} total={100}>
+          {cpu}%
+        </PercentageBar>
+      </TableCell>
+      <TableCell>
+        <PercentageBar num={memoryInfo.rss} total={mem[0]}>
+          {memoryConverter(memoryInfo.rss)}/{memoryConverter(mem[0])}(
+          {(memoryInfo.rss / mem[0]).toFixed(1)}
+          %)
+        </PercentageBar>
+      </TableCell>
+      <TableCell>N/A</TableCell>
+      <TableCell>N/A</TableCell>
+      <TableCell align="center">N/A</TableCell>
+      <TableCell align="center">N/A</TableCell>
+      <TableCell>
+        <Link to={workerLogUrl} target="_blank">
+          Log
+        </Link>
+      </TableCell>
+    </TableRow>
+  );
+};
+
+type NodeRowsProps = {
+  /**
+   * Details of the node
+   */
+  node: NodeDetail;
+  /**
+   * The index of the table. Needed to guarantee key uniqueness.
+   */
+  rowIndex: number;
+  /**
+   * Whether the node row should refresh data about its workers.
+   */
+  isRefreshing: boolean;
+};
+
+/**
+ * The rows related to a node and its workers. Expandable to show information about workers.
+ */
+export const NodeRows = ({ node, rowIndex, isRefreshing }: NodeRowsProps) => {
```

Comment:
can use 'const NodeRows: React.FC<NodeRowsProps> = ({ node, rowIndex, isRefreshing })' instead of 'const NodeRows = ({ node, rowIndex, isRefreshing }: NodeRowsProps)'.
the same as line 47

Reply:
In line 47, we do not use `React.FC` there either.

In general, `React.FC` seems to have been mistakenly believed to be a common and useful pattern because create-react-app's template default to using it. They have since removed that from their template with explanation: https://github.com/facebook/create-react-app/pull/8177